test/emqx_schema.erl

```diff
@@ -0,0 +1,1375 @@
+    , {"dist_listen_min", integer("kernel.inet_dist_listen_min", undefined)}
+    , {"dist_listen_max", integer("kernel.inet_dist_listen_max", undefined)}
+    , {"backtrace_depth", integer("emqx.backtrace_depth", 16)}
+    ];
+
+fields("rpc") ->
+    [ {"mode", fun rpc__mode/1}
+    , {"async_batch_size", integer("gen_rpc.max_batch_size", 256)}
+    , {"port_discovery", fun rpc__port_discovery/1}
+    , {"tcp_server_port", integer("gen_rpc.tcp_server_port", 5369)}
+    , {"tcp_client_num", fun rpc__tcp_client_num/1}
+    , {"connect_timeout", duration("gen_rpc.connect_timeout", "5s")}
+    , {"send_timeout", duration("gen_rpc.send_timeout", "5s")}
+    , {"authentication_timeout", duration("gen_rpc.authentication_timeout", "5s")}
+    , {"call_receive_timeout", duration("gen_rpc.call_receive_timeout", "15s")}
+    , {"socket_keepalive_idle", duration("gen_rpc.socket_keepalive_idle", "7200s")}
+    , {"socket_keepalive_interval", duration("gen_rpc.socket_keepalive_interval", "75s")}
+    , {"socket_keepalive_count", integer("gen_rpc.socket_keepalive_count", 9)}
+    , {"socket_sndbuf", bytesize("gen_rpc.socket_sndbuf", "1MB")}
+    , {"socket_recbuf", bytesize("gen_rpc.socket_recbuf", "1MB")}
+    , {"socket_buffer", bytesize("gen_rpc.socket_buffer", "1MB")}
+    ];
+
+fields("log") ->
+    [ {"to", fun log__to/1}
+    , {"level", fun log__level/1}
+    , {"time_offset", string(undefined, "system")}
+    , {"primary_log_level", fun log__primary_log_level/1}
+    , {"dir", string(undefined,"log")}
+    , {"file", fun log__file/1}
+    , {"chars_limit", integer(undefined, -1)}
+    , {"supervisor_reports", fun log__supervisor_reports/1}
+    , {"max_depth", fun log__max_depth/1}
+    , {"formatter", fun log__formatter/1}
+    , {"single_line", boolean(undefined, true)}
+    , {"rotation", ref("rotation")}
+    , {"size", fun log__size/1}
+    , {"sync_mode_qlen", integer(undefined, 100)}
+    , {"drop_mode_qlen", integer(undefined, 3000)}
+    , {"flush_qlen", integer(undefined, 800)}
+    , {"overload_kill", flag(undefined, true)}
+    , {"overload_kill_mem_size", bytesize(undefined, "30MB")}
+    , {"overload_kill_qlen", integer(undefined, 20000)}
+    , {"overload_kill_restart_after", fun log__overload_kill_restart_after/1}
+    , {"burst_limit", string(undefined, "disabled")}
+    , {"error_logger", fun log__error_logger/1}
+    , {"debug", ref("additional_log_file")}
+    , {"info", ref("additional_log_file")}
+    , {"notice", ref("additional_log_file")}
+    , {"warning", ref("additional_log_file")}
+    , {"error", ref("additional_log_file")}
+    , {"critical", ref("additional_log_file")}
+    , {"alert", ref("additional_log_file")}
+    , {"emergency", ref("additional_log_file")}
+    ];
+
+fields("additional_log_file") ->
+    [ {"file", fun string/1}];
+
+fields("rotation") ->
+    [ {"size", bytesize(undefined, "10MB")}
+    , {"count", integer(undefined, 5)}
+    ];
+
+fields("lager") ->
```

Comment:
Unrelated to the PR, but why do we still have `lager` configuration? I believe this dependency has been removed

Reply:
dependency of dependency